src/pages/ReportDetailsPage.js

```diff
@@ -182,11 +185,8 @@ function ReportDetailsPage(props) {
             <FullPageNotFoundView shouldShow={_.isEmpty(props.report)}>
                 <HeaderWithBackButton
                     title={props.translate('common.details')}
-                    shouldNavigateToTopMostReport
-                    onBackButtonPress={() => {
-                        Navigation.goBack();
-                        Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(props.report.reportID));
-                    }}
+                    shouldNavigateToTopMostReport={!props.route.params.backTo}
+                    onBackButtonPress={() => Navigation.goBack(props.route.params.backTo)}
```

Comment:
@ndiary, do you consider the onBackButtonPress adequately appropriate?

If you don't mind me asking, what was the major issue with Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(props.report.reportID)); ?